packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,207 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { Currency, Locale, Market } from '.';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Represents the cart object for the app checkout flow.
+ *
+ * @beta
+ */
+export interface Cart {
+  /** Version of the cart. */
+  readonly version: CartVersion;
+  /** The id of the cart. */
+  readonly id: string;
+  /** The cart info. */
+  cartInfo: CartInfo;
+  /** The cart items. */
+  cartItems: CartItem[];
+}
+
+/**
+ * Version of the cart.
+ *
+ * @beta
+ */
+interface CartVersion {
+  /** Represents the major version number. */
+  majorVersion: number;
+  /** Represents the minor version number. */
+  minorVersion: number;
+}
+
+/**
+ * Represents the cart information
+ *
+ * @beta
+ */
+interface CartInfo {
+  /** The country market where the products are selling. */
+  readonly market: Market;
+  /** The identifier to tell the cart is checked out by admin or information worker */
+  readonly intent: Intent;
+  /** Locale for the user */
+  readonly locale: Locale;
+  /** The status of the cart. */
+  status: CartStatus;
+  /** Currency code for the cart item price. */
+  readonly currency: Currency;
+}
+
+/**
+ * Represents the basic cart item information.
+ *
+ * @beta
+ */
+interface Item {
+  /** The id of the cart item. */
+  readonly id: string;
+  /** The display name of the cart item. */
+  name: string;
+  /** The quantity of the cart item. */
+  quantity: number;
+  /** The price of the single cart item. */
+  price: number;
+  /** The thumbnail imageURL of the cart item. */
+  readonly imageURL?: string;
+}
+
+/**
+ * Represents the cart item that could have accessories
+ *
+ * @beta
+ */
+export interface CartItem extends Item {
+  /** Accessories to the item if existing. */
+  readonly accessories?: Item[];
+}
+
+/**
+ * Represents the persona creating the cart.
+ *
+ * @beta
+ */
+export enum Intent {
+  /** The cart is created by admin of an organization. */
+  AdminUser = 'AdminUser',
```

Comment:
can this be 
AdminTAC
AdminClient
IWClient 

and this enum will grow, how easy is it to add more to this enum without making much changes to the public interface?
Would it be better to keep it a string?